hazelcast/src/main/java/com/hazelcast/query/impl/QueryableEntry.java

```diff
@@ -116,7 +121,7 @@ private static Object extractAttributeValueIfAttributeQueryConstant(Serializatio
         if (KEY_ATTRIBUTE_NAME.value().equals(attributeName)) {
             return serializationService.toObject(key);
         } else if (THIS_ATTRIBUTE_NAME.value().equals(attributeName)) {
-            return serializationService.toObject(value);
+            return value instanceof Data ? serializationService.toObject(value) : value;
```

Comment:
Yep, but `serializationService` is null in some execution paths - like in map-reduce, since there's never any seralised / portable data there. 
In 3.7 there'll be some further work on the extraction mechanisms and I plan to move it apart from QueryableEntry and encapsulate it.